### ISSUE TITLE
Error while applying for the first time

### ISSUE THREAD

**Issue by xenadmin** [ryancbutler/Citrix#7]
NetScaler 11.0 66.11nc VPX - no HA

Tried to test your script. Got the following error. But I can't see a problem (right now)

```
PS C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1> .\set-nsssl.ps1 -nsip 192.168.23.5 -adminaccount nsroot -adminpassword "Passw0rd" -ciphergroupname "Modern-Ciphers"
Connecting HTTP
Logging in...
Checking for DHKEY:  dhkey2048.key
DHKEY NOT present...
DHKEY creation in process.  Can take a couple of minutes...
Checking for cipher group...
Creating Modern-Ciphers
VPX FOUND...
Creating cipher group for VPX...
Checking for rewrite policy...
Creating  pol-sts-force
Skipping SSL managment IPs...
Checking LB VIPs...
Checking VPN VIPs...
_XD_portal.example.com
Unbinding DEFAULT
Binding Modern-Ciphers
Invoke-RestMethod : { "errorcode": 1647, "message": "Input file(s) not present or not accessible in current partition", "severity": "ERROR" }
In C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1\set-nsssl.ps1:274 Zeichen:5
+     Invoke-RestMethod -uri "$hostname/nitro/v1/config/sslvserver/$Name" -body $b ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
Checking CS VIPs...
Setting SSL renegotiation...
Saving NS config..
Logging out...
PS C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1>
```

**Comment by ryancbutler** [comment]
Is this still an issue for you or did it end up running?

**Comment by xenadmin** [comment]
This is still an issue. This happens everytime I use the script for the first time with a VPX appliance. Just tried I on another instance. But I'm not sure if this error has any downside, or if it works as it should. 

Here is the latest error log, I use the latest version of the script, the line of the error has changed.

```
PS C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1> .\set-nsssl.ps1 -nsip 192.168.25.5 -adminaccount nsroot -adminpassword "xxx" -nosave
Connecting HTTP
Logging in...
Checking for DHKEY:  dhkey2048.key
DHKEY NOT present...
DHKEY creation in process.  Can take a couple of minutes...
Checking for cipher group...
Creating custom-ssllabs-cipher
VPX FOUND...
Creating cipher group for VPX...
Checking for rewrite policy...
Creating  pol-sts-force
Skipping SSL managment IPs...
Checking LB VIPs...
Checking VPN VIPs...
portal.example.com
Unbinding DEFAULT
Binding custom-ssllabs-cipher
Invoke-RestMethod : { "errorcode": 1647, "message": "Input file(s) not present or not accessible in current partition", "severity": "ERROR" }
In C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1\set-nsssl.ps1:277 Zeichen:5
+     Invoke-RestMethod -uri "$hostname/nitro/v1/config/sslvserver/$Name" -body $b ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
Checking CS VIPs...
Setting SSL renegotiation...
NOT saving NS config..
Logging out...
PS C:\Users\Marco.Hofmann\Downloads\NetScaler 11.1>
```

Error says it's line 277 concerning the function #disables sslv2 and sslv3

EDIT1: Found the first downsinde. The DH key params checkbox wasn't set and the dhkey not bound.

EDIT2: Simply starting the script a second time, seams to solve everything that gets left out the first time.

**Comment by ryancbutler** [comment]
I think I had an issue with the error handling where the DHKEY creation might have been timing out but wasn't getting the output.  I removed the "already exists" error handling in the "create" function since it's already in a dedicated "check" function and increased the timeout to 10 minutes.  

Try this when you get a chance.  aeb67d3

**Comment by xenadmin** [comment]
Maybe I have a new VPX Express installation on friday. I will try the new version then.

**Comment by xenadmin** [comment]
The appointment for the VPX Express was today. After the Gateway configuration I tried the new version of the script. The error I reported has been gone. All settings needed for an A+ have been set during the first run, as far as I can see it. 
Only one thing bothered me. The DHKEY generation took only a few seconds. 

```
Logging in...
Checking for DHKEY:  dhkey2048.key
DHKEY NOT present...
DHKEY creation in process.  Can take a couple of minutes...
```

On the two other appliances with the "old" script it took a minute or two, like expected. But this time only 2 seconds. But the file was there and linked to the virtual server. I'm not quite sure what has happend here.
